File: DS/pattern.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    int s = 1, e = n*n+1;
    vector<string> arr;
    for(int i = n; i >= 1; i--){
        string str;
        int j;
        for(j = s; j < i+s; j++)
            str += to_string(j) + '*';
        s = j;
        for(j = e; j < i+e; j++){
            str += to_string(j);
            if(j < i+e-1)
                str += '*';
        }
        e -= (i-1);
        arr.push_back(str);
    }
    for(int i = 0; i < arr.size(); i++){
        string s(i, '-');
        cout << s;
        cout << arr[i] << "\n";
    }
}

File: DS/pattern.py
n = int(input())
s, e = 1, n*n+1
arr = []

for i in range(n, 0, -1):
    str_ = ""
    j = s
    while j < i + s:
        str_ += str(j) + '*'
        j += 1
    s = j
    j = e
    while j < i + e:
        str_ += str(j)
        if j < i + e - 1:
            str_ += '*'
        j += 1
    e -= (i - 1)
    arr.append(str_)

for i in range(len(arr)):
    s = '-' * i
    print(s + arr[i])
